ITT-39: Generate Checkout automatically

michaelpaul opened the pull request:
<!-- 🎉 Thank you for submitting a pull request! 🎉  -->

## Summary
<!--
Describe the changes proposed in this pull request:
- What is the motivation for this change? 
- What existing problem does this pull request solve?
-->
The Checkout service has been fully generated using new models from OpenAPI generator version v6.4.0. The service types have been generated too, but they still use the old style API (templates). 

## Tested scenarios
<!-- Description of tested scenarios -->
- unit tests
- integration tests

Fixes #162

gcatanese commented:
Great work! Here are some comments:

- we could remove the Contact email address from the heading in each file
- in the generated code the method name in the comments doesn't match the actual method name, for example
```
/*
PostCancels Cancel an authorised payment
...
*/
func (a Checkout) CancelAuthorisedPayment(req *CreateStandalonePaymentCancelRequest, ctxs ..._context.Context) (StandalonePaymentCancelResource, *_nethttp.Response, error) {
	
```
- it is a great opportunity to have both unit testing (mock-based) and integration testing for checkout. Also moving all checkout tests in a `checkout` folder
- `api_recurring` new file without tests
- shouldn't `client.mustache` and `configuration.mustache` be removed (they are not used I think)?
- should `CheckoutPaymentMethod` be renamed to `PaymentMethod`?
- README.md is not updated (ie Checkout v70)
- could we also generate the `doc`? It creates useful snippets and references for the developers

michaelpaul commented:
Thanks @gcatanese! I've made some changes:

> * we could remove the Contact email address from the heading in each file

Done.

> * in the generated code the method name in the comments doesn't match the actual method name...

Fixed.

> * it is a great opportunity to have both unit testing (mock-based) and integration testing for checkout. Also moving all checkout tests in a `checkout` folder
> * `api_recurring` new file without tests

Test will be improved on dedicated ticket (ITT-66). 

> * shouldn't `client.mustache` and `configuration.mustache` be removed (they are not used I think)?

Fixed.

> * should `CheckoutPaymentMethod` be renamed to `PaymentMethod`?

The `PaymentMethod` model is already used into the `/paymentMethods` call.

> * README.md is not updated (ie Checkout v70)

Fixed.

> * could we also generate the `doc`? It creates useful snippets and references for the developers

Fixed. I've added model docs only since the API is likely to change soon.

> 
> One note: we need to find a way to automate (or deal with) `model.mustache` that has the library version v6 hardcoded in the imports
> 
> ```
>     "github.com/adyen/adyen-go-api-library/v6/src/common"
> ```

That can be done later when we automate the release process as a whole (ITT-65).
